qemu/tests/boot_with_max_queues.py

```diff
@@ -0,0 +1,238 @@
+        if not guest_name:
+            test.error("Could not get product dirname of the vm")
+        guest_arch = virtio_win.arch_dirname_iso(session)
+        if not guest_arch:
+            test.error("Could not get architecture dirname of the vm")
+
+        pdb_middle_path = "%s\\%s" % (guest_name, guest_arch)
+        pdb_find_cmd = 'dir /b /s %s\\%s.pdb | findstr "\\%s\\\\"'
+        pdb_find_cmd %= (viowin_ltr, driver_name, pdb_middle_path)
+        pdb_path = session.cmd(pdb_find_cmd, timeout=timeout).strip()
+        logging.info("Found Netkvm.pdb file at '%s'", pdb_path)
+        _copy_file(session, pdb_path, "c:\\")
+
+    login_timeout = int(params.get("login_timeout", 360))
+    cmd_timeout = int(params.get("cmd_timeout", 240))
+
+    # boot the vm with the max queues
+    queues = int(params["queues"])
+    error_context.context("Boot the guest with queues = %s" % queues)
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=login_timeout)
+
+    # enable max queues for linux guest
+    if params["os_type"] == "linux":
+        nic = vm.virtnet[0]
+        ifname = utils_net.get_linux_ifname(session, nic.mac)
+        set_queue_cmd = "ethtool -L %s combined %s" % (ifname, queues)
+        status, output = session.cmd_status_output(set_queue_cmd, timeout=cmd_timeout, safe=True)
+        if status and "combined unmodified, ignoring" not in output:
+            test.error("Failed to set queues to %s with status = %s and output= %s" %
+                       (queues, status, output))
+        check_queue_cmd = "ethtool -l %s" % ifname
+        output = session.cmd_output(check_queue_cmd, timeout=cmd_timeout)
+        if len(re.findall(r"Combined:\s+%d\s" % queues, output)) != 2:
+            test.fail("Fail to set queues to %s on %s" %
+                      (queues, nic.nic_name))
+
+        # check the msi for linux guest
+        error_context.context("Check the msi number in guest", logging.info)
+        devices = session.cmd_output("lspci | grep Eth", timeout=cmd_timeout, safe=True).strip()
+        for device in devices.split("\n"):
+            if not device:
+                continue
+            d_id = device.split()[0]
+            msi_check_cmd = params["msi_check_cmd"] % d_id
+            status, output = session.cmd_status_output(msi_check_cmd, timeout=cmd_timeout, safe=True)
+            find_result = re.search(r"MSI-X: Enable\+\s+Count=(\d+)", output)
+            if "pci=nomsi" in output or not find_result:
+                test.fail("No MSI info in output: %s" % output)
+            msis = int(find_result.group(1))
+            if msis != 2 * queues + 2:
+                test.fail("MSI not correct with output: %s" % output)
+    else:
+        # check the msi for windows guest with trace view
+        error_context.context("Check the msi number in guest", logging.info)
+        session = prepare_traceview_windows(session)
```

Comment:
@kimi1978wy Since you dropped `retrun session` from `def prepare_traceview_windows()`, this line need to be modified.
```suggestion
        prepare_traceview_windows(session)
```